src/annoylib.h

```diff
@@ -230,7 +232,7 @@ struct Hamming {
 
   template<typename T>
   static inline T pq_distance(T distance, T margin, int child_nr) {
-    return distance - (margin != child_nr);
+    return distance - (margin != (unsigned int) child_nr);
```

Comment:
think this could be just `(T)child_nr` but not blocking

Reply:
That's a good one.

Reply:
Tested, and now committed, here.

Reply:
i'm planning to change the splitting code for Hamming distance anyway, so not important